diff --git a/ops/third_party/chainlink-common-keystore/corekeys/starkkey/curve_compat.go b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/curve_compat.go
new file mode 100644
index 00000000..e96984f7
--- /dev/null
+++ b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/curve_compat.go
@@ -0,0 +1,12 @@
+package starkkey
+
+import "math/big"
+
+// STARK curve subgroup order.
+var curveOrder = func() *big.Int {
+	n, ok := new(big.Int).SetString("3618502788666131213697322783095070105526743751716087489154079457884512865583", 10)
+	if !ok {
+		panic("invalid stark curve order")
+	}
+	return n
+}()
diff --git a/ops/third_party/chainlink-common-keystore/corekeys/starkkey/export.go b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/export.go
new file mode 100644
index 00000000..a3b06ee2
--- /dev/null
+++ b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/export.go
@@ -0,0 +1,46 @@
+package starkkey
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+
+	commonkeystore "github.com/smartcontractkit/chainlink-common/keystore"
+	"github.com/smartcontractkit/chainlink-common/keystore/internal"
+)
+
+const keyTypeIdentifier = "StarkNet"
+
+// FromEncryptedJSON gets key from json and password
+func FromEncryptedJSON(keyJSON []byte, password string) (Key, error) {
+	return internal.FromEncryptedJSON(
+		keyTypeIdentifier,
+		keyJSON,
+		password,
+		adulteratedPassword,
+		func(_ internal.EncryptedKeyExport, rawPrivKey internal.Raw) (Key, error) {
+			return KeyFor(rawPrivKey), nil
+		},
+	)
+}
+
+// ToEncryptedJSON returns encrypted JSON representing key
+func ToEncryptedJSON(key Key, password string, scryptParams commonkeystore.ScryptParams) (export []byte, err error) {
+	return internal.ToEncryptedJSON(
+		keyTypeIdentifier,
+		key,
+		password,
+		scryptParams.N,
+		scryptParams.P,
+		adulteratedPassword,
+		func(id string, key Key, cryptoJSON keystore.CryptoJSON) internal.EncryptedKeyExport {
+			return internal.EncryptedKeyExport{
+				KeyType:   id,
+				PublicKey: key.StarkKeyStr(),
+				Crypto:    cryptoJSON,
+			}
+		},
+	)
+}
+
+func adulteratedPassword(password string) string {
+	return "starkkey" + password
+}
diff --git a/ops/third_party/chainlink-common-keystore/corekeys/starkkey/key.go b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/key.go
new file mode 100644
index 00000000..47ba5f0e
--- /dev/null
+++ b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/key.go
@@ -0,0 +1,87 @@
+package starkkey
+
+import (
+	crypto_rand "crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/starknet.go/curve"
+
+	"github.com/smartcontractkit/chainlink-common/keystore/internal"
+	adapters "github.com/smartcontractkit/chainlink-common/pkg/loop/adapters/starknet"
+)
+
+func KeyFor(raw internal.Raw) Key {
+	k := Key{raw: raw}
+
+	priv := new(big.Int).SetBytes(internal.Bytes(raw))
+	k.signFn = func(hash *big.Int) (x, y *big.Int, err error) {
+		return curve.Sign(hash, priv)
+	}
+	k.pub.X, k.pub.Y = curve.PrivateKeyToPoint(priv)
+	return k
+}
+
+type PublicKey struct {
+	X, Y *big.Int
+}
+
+// Key represents Starknet key
+type Key struct {
+	raw    internal.Raw
+	signFn func(*big.Int) (x, y *big.Int, err error)
+	pub    PublicKey
+}
+
+// New creates new Key
+func New() (Key, error) {
+	return newFrom(crypto_rand.Reader)
+}
+
+// MustNewInsecure return Key if no error
+func MustNewInsecure(reader io.Reader) Key {
+	key, err := newFrom(reader)
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
+
+func newFrom(reader io.Reader) (Key, error) {
+	return GenerateKey(reader)
+}
+
+// ID gets Key ID
+func (key Key) ID() string {
+	return key.StarkKeyStr()
+}
+
+// StarkKeyStr is the starknet public key associated to the private key
+// it is the X component of the ECDSA pubkey and used in the deployment of the account contract
+// this func is used in exporting it via CLI and API
+func (key Key) StarkKeyStr() string {
+	return new(felt.Felt).SetBytes(key.pub.X.Bytes()).String()
+}
+
+// Raw from private key
+func (key Key) Raw() internal.Raw { return key.raw }
+
+func (key Key) Sign(hash []byte) ([]byte, error) {
+	starkHash := new(big.Int).SetBytes(hash)
+	x, y, err := key.signFn(starkHash)
+	if err != nil {
+		return nil, fmt.Errorf("error signing data with curve: %w", err)
+	}
+	sig, err := adapters.SignatureFromBigInts(x, y)
+	if err != nil {
+		return nil, err
+	}
+	return sig.Bytes()
+}
+
+// PublicKey copies public key object
+func (key Key) PublicKey() PublicKey {
+	return key.pub
+}
diff --git a/ops/third_party/chainlink-common-keystore/corekeys/starkkey/ocr2key.go b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/ocr2key.go
new file mode 100644
index 00000000..007c9bfc
--- /dev/null
+++ b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/ocr2key.go
@@ -0,0 +1,188 @@
+package starkkey
+
+import (
+	"bytes"
+	"io"
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/starknet.go/curve"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/chains/evmutil"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-common/keystore/internal"
+)
+
+var _ types.OnchainKeyring = &OCR2Key{}
+
+type OCR2Key struct {
+	Key
+}
+
+func NewOCR2Key(material io.Reader) (*OCR2Key, error) {
+	k, err := GenerateKey(material)
+
+	return &OCR2Key{k}, err
+}
+
+func (sk *OCR2Key) PublicKey() types.OnchainPublicKey {
+	ans := new(felt.Felt).SetBytes(sk.pub.X.Bytes()).Bytes()
+	return ans[:]
+}
+
+func ReportToSigData(reportCtx types.ReportContext, report types.Report) (*big.Int, error) {
+	var dataArray []*big.Int
+
+	rawReportContext := rawReportContext(reportCtx)
+	dataArray = append(dataArray, new(big.Int).SetBytes(rawReportContext[0][:]))
+	dataArray = append(dataArray, new(big.Int).SetBytes(rawReportContext[1][:]))
+	dataArray = append(dataArray, new(big.Int).SetBytes(rawReportContext[2][:]))
+
+	// split report into separate felts for hashing
+	splitReport, err := splitReport(report)
+	if err != nil {
+		return &big.Int{}, err
+	}
+	for i := range splitReport {
+		dataArray = append(dataArray, new(big.Int).SetBytes(splitReport[i]))
+	}
+
+	return curve.ComputeHashOnElements(dataArray), nil
+}
+
+func (sk *OCR2Key) Sign(reportCtx types.ReportContext, report types.Report) ([]byte, error) {
+	hash, err := ReportToSigData(reportCtx, report)
+	if err != nil {
+		return []byte{}, err
+	}
+	r, s, err := sk.signFn(hash)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// enforce s <= N/2 to prevent signature malleability
+	if s.Cmp(new(big.Int).Rsh(curveOrder, 1)) > 0 {
+		s.Sub(curveOrder, s)
+	}
+
+	// encoding: public key (32 bytes) + r (32 bytes) + s (32 bytes)
+	buff := bytes.NewBuffer([]byte(sk.PublicKey()))
+	if _, err := buff.Write(padBytes(r.Bytes())); err != nil {
+		return []byte{}, err
+	}
+	if _, err := buff.Write(padBytes(s.Bytes())); err != nil {
+		return []byte{}, err
+	}
+
+	out := buff.Bytes()
+	if len(out) != sk.MaxSignatureLength() {
+		return []byte{}, errors.Errorf("unexpected signature size, got %d want %d", len(out), sk.MaxSignatureLength())
+	}
+	return out, nil
+}
+
+func (sk *OCR2Key) Sign3(digest types.ConfigDigest, seqNr uint64, r types.Report) (signature []byte, err error) {
+	return nil, errors.New("not implemented")
+}
+
+func (sk *OCR2Key) SignBlob(b []byte) ([]byte, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (sk *OCR2Key) Verify(publicKey types.OnchainPublicKey, reportCtx types.ReportContext, report types.Report, signature []byte) bool {
+	// check valid signature length
+	if len(signature) != sk.MaxSignatureLength() {
+		return false
+	}
+
+	// convert OnchainPublicKey (starkkey) into ecdsa public keys (prepend 2 or 3 to indicate +/- Y coord)
+	var keys [2]PublicKey
+	keys[0].X = new(big.Int).SetBytes(publicKey)
+	yFelt := curve.GetYCoordinate(new(felt.Felt).SetBigInt(keys[0].X))
+	if yFelt == nil {
+		return false
+	}
+	keys[0].Y = yFelt.BigInt(new(big.Int))
+
+	keys[1].X = keys[0].X
+	keys[1].Y = new(big.Int).Neg(keys[0].Y)
+
+	hash, err := ReportToSigData(reportCtx, report)
+	if err != nil {
+		return false
+	}
+
+	r := new(big.Int).SetBytes(signature[32:64])
+	s := new(big.Int).SetBytes(signature[64:])
+
+	// Only allow canonical signatures to avoid signature malleability. Verify s <= N/2
+	if s.Cmp(new(big.Int).Rsh(curveOrder, 1)) == 1 {
+		return false
+	}
+
+	ok, err := curve.Verify(hash, r, s, keys[0].X)
+	return err == nil && ok
+}
+
+func (sk *OCR2Key) Verify3(publicKey types.OnchainPublicKey, cd types.ConfigDigest, seqNr uint64, r types.Report, signature []byte) bool {
+	return false
+}
+
+func (sk *OCR2Key) VerifyBlob(pubkey types.OnchainPublicKey, b, sig []byte) bool {
+	return false
+}
+
+func (sk *OCR2Key) MaxSignatureLength() int {
+	return 32 + 32 + 32 // publickey + r + s
+}
+
+func (sk *OCR2Key) Marshal() ([]byte, error) {
+	return padBytes(internal.Bytes(sk.raw)), nil
+}
+
+// https://github.com/NethermindEth/juno/blob/3e71279632d82689e5af03e26693ca5c58a2376e/pkg/crypto/weierstrass/weierstrass.go#L377
+const privateKeyLen = 32
+
+func (sk *OCR2Key) Unmarshal(in []byte) error {
+	// enforce byte length
+	if len(in) != privateKeyLen {
+		return errors.Errorf("unexpected seed size, got %d want %d", len(in), privateKeyLen)
+	}
+
+	sk.Key = KeyFor(internal.NewRaw(in))
+	return nil
+}
+
+func splitReport(report types.Report) ([][]byte, error) {
+	chunkSize := 32
+	if len(report)%chunkSize != 0 {
+		return [][]byte{}, errors.New("invalid report length")
+	}
+
+	// order is guaranteed by buildReport:
+	//   observation_timestamp
+	//   observers
+	//   observations_len
+	//   observations
+	//   juels_per_fee_coin
+	//   gas_price
+	slices := [][]byte{}
+	for i := 0; i < len(report)/chunkSize; i++ {
+		idx := i * chunkSize
+		slices = append(slices, report[idx:(idx+chunkSize)])
+	}
+
+	return slices, nil
+}
+
+// NOTE: this should sit in the ocr2 package but that causes import cycles
+func rawReportContext(repctx types.ReportContext) [3][32]byte {
+	rawReportContext := evmutil.RawReportContext(repctx)
+	// NOTE: Ensure extra_hash is 31 bytes with first byte blanked out
+	// libocr generates a 32 byte extraHash but we need to fit it into a felt
+	rawReportContext[2][0] = 0
+	return rawReportContext
+}
diff --git a/ops/third_party/chainlink-common-keystore/corekeys/starkkey/utils.go b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/utils.go
new file mode 100644
index 00000000..f698afcd
--- /dev/null
+++ b/ops/third_party/chainlink-common-keystore/corekeys/starkkey/utils.go
@@ -0,0 +1,47 @@
+package starkkey
+
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+	"math/big"
+
+	"github.com/NethermindEth/starknet.go/curve"
+
+	"github.com/smartcontractkit/chainlink-common/keystore/internal"
+)
+
+// reimplements parts of
+// https://github.com/NethermindEth/starknet.go/blob/0bdaab716ce24a521304744a8fbd8e01800c241d/curve/curve.go#L702
+// generate the PK as a pseudo-random number in the interval [1, CurveOrder - 1]
+// using io.Reader, and Key struct
+func GenerateKey(material io.Reader) (k Key, err error) {
+	max := new(big.Int).Sub(curveOrder, big.NewInt(1))
+
+	priv, err := rand.Int(material, max)
+	if err != nil {
+		return k, err
+	}
+	k.signFn = func(hash *big.Int) (x, y *big.Int, err error) {
+		return curve.Sign(hash, priv)
+	}
+
+	k.pub.X, k.pub.Y = curve.PrivateKeyToPoint(priv)
+	if k.pub.X == nil || k.pub.Y == nil {
+		return k, errors.New("key gen is not on stark curve")
+	}
+	k.raw = internal.NewRaw(padBytes(priv.Bytes()))
+
+	return k, nil
+}
+
+// pad bytes to privateKeyLen
+func padBytes(a []byte) []byte {
+	if len(a) < privateKeyLen {
+		pad := make([]byte, privateKeyLen-len(a))
+		return append(pad, a...)
+	}
+
+	// return original if length is >= to specified length
+	return a
+}
